webpack.config.js

```diff
@@ -84,6 +98,60 @@ const generateConfig = ({extensionPath, devMode=false, customOutputPath, analyze
       path.resolve(__dirname, "dist");
   utils.verbose(`Webpack writing output to: ${outputPath}`);
 
+  /**
+   * Here we put the libraries that are unlikely to change for a long time.
+   * Every change or update of any of these libraries will change the hash
+   * of the big vendor bundle, so we must be sure they're stable both internally
```

Comment:
When this needs changing, the bundle hash will change and so will the script src (in `dist/index.html`), correct? And this means that the client will treat it as completely different request, and thus there's no danger of using the old, now invalid bundle from the cache?

If this is true, is there benefit to adding a (large) `max-age` to the response for these bundles, since if their contents change the hash will also change?

Reply:
Yes all of the above is correct, I didn't check how you're caching .js files atm on the public site but basically the whole point of this setup is you set index.html to be served with no no cache whatsoever (something like `no-cache,no-store,must-revalidate`), then you set nice and long (multiple months) caching for the .js part, and the hashes in the names of the scripts take care of cache busting for you. 

And when you build the index.html will generate automatically with the correct hashes, modifying only those that should be modified.

Reply:
Great! While nextstrain.org uses a custom server, I know there are other groups using `auspice view` as a production server, so we should set these here. Playing around with it, it _seems_ as simple as 

```diff
- res.sendFile(path.join(auspiceBuild.baseDir, "dist/index.html"))
+res.sendFile(path.join(auspiceBuild.baseDir, "dist/index.html"), {headers: {"Cache-Control": "no-cache, no-store, must-revalidate"}});
```
and 
```diff
- app.use("/dist", expressStaticGzip(auspiceBuild.distDir));
+ app.use("/dist", expressStaticGzip(auspiceBuild.distDir, {maxAge: '365d'}));
```

But wanted to get your thoughts here, as the nextstrain.org server is similar to the auspice server here.

Reply:
Sorry for forgetting to reply here, yep it should suffice as far as I can tell